modules/bloxone_infra_host_azure/variables.tf

```diff
@@ -0,0 +1,124 @@
+variable "join_token" {
+  description = "The join token to use for the BloxOne Host. If not provided, a join token will be created."
+  type        = string
+  default     = null
+}
+
+variable "vm_name" {
+  description = "The name of the virtual machine"
+  type        = string
+}
+
+variable "resource_group_name" {
+  description = "The name of the resource group in which the resources will be created"
+  type        = string
+}
+
+variable "location" {
+  description = "The location where the resources will be created"
+  type        = string
+}
+
+variable "vm_size" {
+  description = "Size of the Virtual Machine based on Azure sizing"
+  type        = string
+  default     = "Standard_F8s"
+}
+
+variable "vm_network_interface_ids" {
+  description = "The network interface ids that will be associated to the BloxOne Host"
+  type        = list(string)
+}
+
+variable "tags" {
+  description = "The tags to use for the BloxOne Host."
+  type        = map(string)
+  default     = {}
+}
+
+variable "azure_instance_tags" {
+  description = "The tags to use for the Azure virtual machine."
+  type        = map(string)
+  default     = {}
+}
+
+variable "os_disk_caching" {
+  description = "The caching type to use for the OS disk."
+  type        = string
+  default     = "ReadWrite"
+}
+
+variable "os_disk_storage_account_type" {
+  description = "The storage account type to use for the OS disk."
+  type        = string
+  default     = "Standard_LRS"
+}
+
+variable "source_image_reference_publisher" {
+  description = "The publisher of the image that you want to deploy"
+  type        = string
+  default     = "infoblox"
+}
+
+variable "source_image_reference_offer" {
+  description = "The offer of the image that you want to deploy"
+  type        = string
+}
+
+variable "source_image_reference_sku" {
+  description = "The sku of the image that you want to deploy"
+  type        = string
+  default     = "infoblox-bloxone"
+}
+
+variable "source_image_reference_version" {
+  description = "The version of the image that you want to deploy."
+  type        = string
+}
```

Comment:
Having a default value here might be good. Note that the module doesn't work for the image 3.5 because it doesn't support tags in cloud-config. Later we will have to do some validations around it.